### PR TITLE
Introduce Json schema checks

### PR DESCRIPTION
**Pull request by flagbird**

### CODE REVIEW

File: src/EventListener/SavePropertyListener.php [hunk 1]
```diff
@@ -50,6 +55,10 @@ public function onCategoryPostSave(GenericEvent $event): void
             return;
         }
 
+        if ($this->validator->validate($properties) !== []) {
+            throw new RuntimeException('Invalid properties format');
```
Comment (pascal-zarrad) [inline src/EventListener/SavePropertyListener.php]: `RuntimeException` is very generic, maybe consider using or creating a more specific exception type.

File: src/Schema/SchemaValidator.php [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+ */
+class SchemaValidator
+{
+    private object $schema;
+
+    public function __construct(string $schemaPath)
+    {
+        $realpath = realpath(sprintf('%s/%s', __DIR__, $schemaPath));
+        if ($realpath === false) {
+            throw new LogicException(sprintf('File %s not found', $schemaPath));
```
Comment (pascal-zarrad) [inline src/Schema/SchemaValidator.php]: `LogicException` is very generic, maybe consider using or creating a more specific exception type.

**Review by pascal-zarrad — APPROVED** [review]
LGTM 👍
Added two comments for optional changes that may or may not be done.

File: src/EventListener/SavePropertyListener.php (hunk 1 above)
Comment (flagbird) [reply src/EventListener/SavePropertyListener.php]: I've added new Exception classes according to the coding style.

File: src/Schema/SchemaValidator.php (hunk 2 above)
Comment (flagbird) [reply src/Schema/SchemaValidator.php]: I've added new Exception classes according to the coding style.

**Review by pascal-zarrad — APPROVED** [review]
LGTM 👍

**Pull request merged**